crates/oxc_linter/src/rules/security/api_keys/mod.rs

```diff
@@ -174,4 +204,27 @@ impl Rule for ApiKeys {
             }
         }
     }
+
+    fn from_configuration(value: Value) -> Self {
+        let Some(obj) = value.get(0) else {
+            return Self::default();
+        };
+        let config = serde_json::from_value::<ApiKeysConfig>(obj.clone()).unwrap();
+
+        // TODO: Check if this is worth optimizing, then do so if needed.
+        let mut rules = ALL_RULES.clone();
+        rules.extend(config.custom_patterns.into_iter().map(|pattern| {
+            let regex = Regex::new(&pattern.pattern).unwrap();
```

Comment:
should we use `.expect()` here with a custom message, so we can let the user know that their regex pattern was invalid?